# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/ServiceWorkerGlobalScope.webidl
# Commit: 9c8b6e35bf3f
# Full Hash: 9c8b6e35bf3f9916e114ced0473be4b778c7bab1
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 1 - CookieStore API - WebIDL, r=smaug,edgul
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215144
# ==============================================================================

diff -r a9b3b52f33a5 -r 9c8b6e35bf3f dom/webidl/ServiceWorkerGlobalScope.webidl
--- a/dom/webidl/ServiceWorkerGlobalScope.webidl	Tue Sep 10 16:21:33 2024 +0000
+++ b/dom/webidl/ServiceWorkerGlobalScope.webidl	Tue Sep 10 16:31:38 2024 +0000
@@ -7,6 +7,7 @@
  * http://slightlyoff.github.io/ServiceWorker/spec/service_worker/index.html
  * http://w3c.github.io/push-api/
  * https://notifications.spec.whatwg.org/
+ * https://wicg.github.io/cookie-store/#ServiceWorkerGlobalScope
  *
  * You are granted a license to use, reproduce and create derivative works of
  * this document.
@@ -47,3 +48,13 @@
 // Mixin the WebExtensions API globals (the actual properties are only available to
 // extension service workers, locked behind a Func="extensions::ExtensionAPIAllowed" annotation).
 ServiceWorkerGlobalScope includes ExtensionGlobalsMixin;
+
+// https://wicg.github.io/cookie-store/#ServiceWorkerGlobalScope
+partial interface ServiceWorkerGlobalScope {
+  [SameObject, Pref="dom.cookieStore.enabled"] readonly attribute CookieStore cookieStore;
+
+/* Bug 1475599 - We decide to do not implement the entire cookie-store spec.
+ * Instead, we implement only the subset that is compatible with document.cookie
+ attribute EventHandler oncookiechange;
+ */
+};